emulator/ecard.py

```diff
@@ -357,16 +357,16 @@ def cmd_derive(self, nonce=REQUIRED, path=REQUIRED, **unused):
             assert all_hardened(path), 'must have all hardened components'
 
             # auth required, but for TS case only
-            ses_key = self._validate_cvc('derive', unused['epubkey'], unused['xcvc'])
+            self._validate_cvc('derive', unused['epubkey'], unused['xcvc'])
 
             self.cur_slot.save_derive(path)
 
+            self._new_nonce()
+
             msg = b'OPENDIME' + self.nonce + nonce + self.cur_slot.deriv_chain_code
             sig = ec_sig_from_digest(self.cur_slot.privkey, sha256s(msg), EC_FLAG_ECDSA)
             master_pubkey = ec_public_key_from_private_key(self.cur_slot.master_pk)
 
-
-            self._new_nonce()
```

Comment:
bug: nonce used for `msg` is droped & regenerated just before response is returned, then caller gets new card nonce and is not able to reconstruct proper `msg` (card nonce is invalid) 

fix: generate new nonce before creating `msg`

Reply:
to be clear, this corrects a difference between emulator and real card.

Reply:
this is a dummy PR, opened against my fork, idk if this makes it same - no access to java src